From 17c079bdda833097004457c63efc7935fdc9b44b Mon Sep 17 00:00:00 2001
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Thu, 23 Sep 2021 11:51:33 +0800
Subject: [PATCH 2883/4585] [Huawei] PCI: add quirk for HiSilicon 5887 device

Offering: HULK
hulk inclusion
category: feature
bugzilla: 13666

--------------------------------

Add PCI quirk for HiSilicon 5887 device.

Signed-off-by: yanbo <joey.yanbo@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>

Signed-off-by: Chen Jun <chenjun102@huawei.com>
Patchwork Links: http://patchwork.huawei.com/project/hulk5.10/list/?series=19012
Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
---
 drivers/pci/quirks.c    | 7 ++++++-
 include/linux/pci_ids.h | 1 +
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 26b526ed90a8..f127f360bfe4 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -5838,6 +5838,8 @@ DECLARE_PCI_FIXUP_EARLY(PCI_VENDOR_ID_HUAWEI, PCIE_DEVICE_ID_HISI_5891,
 			pci_quirk_hisi_fixup_class);
 DECLARE_PCI_FIXUP_EARLY(PCI_VENDOR_ID_HUAWEI, PCIE_DEVICE_ID_HISI_5896,
 			pci_quirk_hisi_fixup_class);
+DECLARE_PCI_FIXUP_EARLY(PCI_VENDOR_ID_HUAWEI, PCIE_DEVICE_ID_HISI_5887,
+			pci_quirk_hisi_fixup_class);
 
 static void pci_quirk_hisi_fixup_bar(struct pci_dev *dev)
 {
@@ -5854,7 +5856,8 @@ static void pci_quirk_hisi_fixup_bar(struct pci_dev *dev)
 		dev->device == PCIE_DEVICE_ID_HISI_8855)
 		start = 1;
 
-	if (dev->device == PCIE_DEVICE_ID_HISI_5991)
+	if (dev->device == PCIE_DEVICE_ID_HISI_5991 ||
+		dev->device == PCIE_DEVICE_ID_HISI_5887)
 		start = 2;
 
 	if (dev->device == PCIE_DEVICE_ID_HISI_5896 ||
@@ -5896,3 +5899,5 @@ DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_HUAWEI, PCIE_DEVICE_ID_HISI_8855,
 			 pci_quirk_hisi_fixup_bar);
 DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_HUAWEI, PCIE_DEVICE_ID_HISI_5991,
 			 pci_quirk_hisi_fixup_bar);
+DECLARE_PCI_FIXUP_HEADER(PCI_VENDOR_ID_HUAWEI, PCIE_DEVICE_ID_HISI_5887,
+			 pci_quirk_hisi_fixup_bar);
diff --git a/include/linux/pci_ids.h b/include/linux/pci_ids.h
index 4c2c58f9d8fa..1f58f5edca05 100644
--- a/include/linux/pci_ids.h
+++ b/include/linux/pci_ids.h
@@ -2583,6 +2583,7 @@
 #define PCIE_DEVICE_ID_HISI_5981	0x5981
 #define PCIE_DEVICE_ID_HISI_8855	0x8855
 #define PCIE_DEVICE_ID_HISI_5991	0x5991
+#define PCIE_DEVICE_ID_HISI_5887	0x5887
 
 /* Hisilicon PCIe NP devices */
 #define PCIE_DEVICE_ID_HISI_5896        0x5896
-- 
2.17.1

